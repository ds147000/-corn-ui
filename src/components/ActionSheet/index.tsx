/* eslint-disable import/export */

import ActionSheet  from './actionSheet'

export { ActionSheetItem } from './item'
export default ActionSheet
