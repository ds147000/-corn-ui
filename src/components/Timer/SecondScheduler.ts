const DEFAULT_INTERVAL = 1000
const MIN_SCE = 60000
const QUEUE_EMPTY_LEN = 0

export type SecondSchedulerDestotry = () => void
export type SecondSchedulerCb = (number: number) => void

export class SecondScheduler {
  private queue = new Set<SecondSchedulerCb>()
  private timeId: NodeJS.Timeout | null = null
  private interval = DEFAULT_INTERVAL

  constructor(interval?: number) {
    if (interval) this.interval = interval
  }

  /** 添加定时任务 */
  add(cb: SecondSchedulerCb): SecondSchedulerDestotry {
    this.queue.add(cb)
    this.create()

    return (): void => {
      this.queue.delete(cb)
      this.remove()
    }
  }

  /** 启动定时器 */
  private create(): void {
    if (this.timeId !== null) return
    this.start()
  }

  /** 定时器回调 */
  private start(): void {
    if (this.timeId) clearTimeout(this.timeId)
    if (this.queue.size === QUEUE_EMPTY_LEN) return

    this.timeId = setTimeout(() => {
      this.emit()
      this.start()
    }, this.interval)
  }

  /** 执行 */
  private emit(): void {
    this.queue.forEach((item) => item(this.interval))
  }

  /** 销毁 */
  private remove(): void {
    if (this.queue.size === QUEUE_EMPTY_LEN) {
      if (this.timeId) clearTimeout(this.timeId)
      this.timeId = null
    }
  }

  destory(): void {
    this.queue.clear()
    this.remove()
  }

}

/** 一秒调度器 */
export const SecondSchedulerApp = new SecondScheduler()

/** 一分钟调度器 */

export const MinuteSchedulerApp = new SecondScheduler(MIN_SCE)

