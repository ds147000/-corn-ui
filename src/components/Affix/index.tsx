/* eslint-disable no-unreachable */
/* eslint-disable no-magic-numbers */
// #if _APP === 'weapp'
import Taro, { usePageScroll } from '@tarojs/taro'
// #endif
import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
import { View } from '@tarojs/components'
import ClassNames from 'classnames'
import { ViewProps } from '../../types/View'
import { judge } from './utils/judge'

export interface AffixProps extends ViewProps {
  position?: 'top' | 'bottom' | 'right' | 'left'
  onChange?(fixed: boolean): void
}

const Affix: React.FC<AffixProps> = ({ children, position = 'top', onChange, className, ...props }) => {
  const [ fixed, setFixed ] = useState(false)
  const [ boxStyle, setBoxStyle ] = useState<React.CSSProperties>({})
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const ref = useRef<any>()

  const oldFixed = useRef<boolean>(false)

  const _class = useMemo(() => {
    return ClassNames('corn-affix', fixed && `corn-affix-${position}`, className)
  }, [ fixed, position, className ])

  const changeFixed = useCallback((value: boolean): void => {
    setFixed((state) => {
      oldFixed.current = state
      return value
    })
  }, [])

  // #if _APP === 'weapp'
  const system = useRef(Taro.getSystemInfoSync())
  const timer = useRef<NodeJS.Timeout>()

  const checkFlexd = (): void => {
    const node = ref.current

    const rootRect = {
      top: 0,
      left: 0,
      bottom: system.current.windowHeight,
      right: system.current.windowWidth
    } as DOMRectReadOnly

    Taro.createSelectorQuery()
      .select('#' + node.uid)
      .fields({ size: true, rect: true }, (res) => {
        if (!res) return

        setBoxStyle({ height: res?.height, width: res?.width })
        const result = judge(res as unknown as DOMRect, rootRect, position, 0)
        changeFixed(result)
      })
      .exec()
  }

  usePageScroll(() => {
    if (timer.current) clearTimeout(timer.current)
    timer.current = setTimeout(() => checkFlexd(), 10)
  })

  useEffect(() => {
    return (): void => {
      if (timer.current) clearTimeout(timer.current)
    }
  }, [])

  // #else
  useEffect(() => {
    // eslint-disable-next-line no-unreachable
    const el: HTMLDivElement = ref.current
    const rootBounds = {
      x: 0,
      y: 0,
      width: window.innerWidth,
      height: window.innerHeight,
      top: 0,
      left: 0,
      bottom: window.innerHeight,
      right: window.innerWidth
    } as DOMRectReadOnly

    setBoxStyle({ height: el.offsetHeight, width: el.offsetWidth })

    function handleScroll(): void {
      const item = el.getBoundingClientRect()
      const result = judge(item, rootBounds, position, 0)
      changeFixed(result)
    }
    handleScroll()
    window.addEventListener('scroll', handleScroll)
    return (): void => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [ position, changeFixed ])
  // #endif

  useEffect(() => {
    if (oldFixed.current !== fixed) {
      oldFixed.current = fixed
      onChange?.(fixed)
    }
  }, [ fixed, onChange ])

  return (
    <View style={boxStyle} ref={ref}>
      <View className={_class} {...props} >
        {children}
      </View>
    </View>
  )
}

export default Affix
