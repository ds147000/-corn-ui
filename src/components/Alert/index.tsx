import React from 'react'
import { View } from '@tarojs/components'
import classNames from 'classnames'
import { ViewProps } from '../../types/View'
import Icon from '../Icon'

const Alter: React.FC<ViewProps> = ({ children, className, ...props }) => {
  const _class = classNames(className, 'corn-alert', 'corn-f')
  return (
    <View className={_class} {...props} >
      <Icon name="error" className="icon" />
      {children}
    </View>
  )
}

export default Alter
