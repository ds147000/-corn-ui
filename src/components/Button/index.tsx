import React, { useMemo } from 'react'
import ClassNames from 'classnames'
import { Button as TaroButton } from '@tarojs/components'
import Link, { LinkProps } from '../Link'
import { ITouchEvent } from '../../types'

/** 按钮大小，max: 最大， big: 超大，large：大，middle：中，small：小，mini */
export type ButtonSize = 'max' | 'big' | 'large' | 'middle' | 'small' | 'mini'
export type ButtonType = 'primary' | 'warn' | 'error' | 'link' | 'pop' | 'default' | 'light' | 'stop'

export interface ButtonProps extends LinkProps {
  /** 按钮大小，max: 最大， big: 超大，large：大，middle：中，small：小，mini */
  size?: ButtonSize
  /** 按钮风格类型 */
  type?: ButtonType
  /** 图标 */
  icon?: React.ReactNode
  /** 将按钮宽度调整为其父宽度的选项	 */
  block?: boolean
  /** 镂空风格 */
  ghost?: boolean
  /** 是否禁用 */
  disabled?: boolean
  /** 跳转链接，填写启用A链接行为 */
  to?: string
  /** 表单点击行为 */
  formType?: 'submit' | 'reset'
  /** 开启固定宽度 */
  auto?: boolean
  /** 点击事件 */
  onClick?(e: ITouchEvent): void
}

const Button: React.FC<ButtonProps> = ({
  size = 'middle',
  type = 'primary',
  block,
  ghost,
  disabled,
  to,
  onClick,
  icon,
  children,
  className,
  formType,
  auto,
  ...props
}) => {
  const isIcon = icon !== undefined
  const isChildren = children !== undefined

  const _class = useMemo(() => {
    return ClassNames(
      'corn-btn',
      'corn-if corn-ac corn-jc',
      `corn-btn-${size}`,
      `corn-btn-${type}`,
      {
        'corn-btn-block': block,
        'corn-btn-ghost': ghost,
        'corn-btn-icon': isIcon && !isChildren,
        'corn-btn-child-icon': isIcon && isChildren,
        'corn-btn-disabled': disabled,
        'corn-btn-auto': auto
      },
      className
    )
  }, [ size, type, block, ghost, disabled, className, auto, isIcon, isChildren ])

  const _onClick = (e: ITouchEvent): void => {
    if (disabled) return

    onClick?.(e)
  }



  if (to) {
    return (
      <Link {...props} target="Text" to={to} className={_class} onClick={_onClick} disabled={disabled} type="default" >
        {children}
        {icon}
      </Link>
    )
  }

  return (
    <TaroButton {...props} formType={formType} className={_class} onClick={_onClick} >
      {children}
      {icon}
    </TaroButton>
  )
}

export default Button
