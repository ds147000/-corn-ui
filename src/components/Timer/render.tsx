import React from 'react'
import { View } from '@tarojs/components'
import type { TimeRenderItem } from './index'

export const DefualtRenderItem: TimeRenderItem = (type, value, last): JSX.Element => {
  if (type === 'day') {
    return (
      <>
        <View className="xrk-f xrk-ac xrk-jc xrk-timer-day" data-testid={type} >
          {value}å¤©
        </View>
      </>
    )
  }

  if (last) {
    return (
      <View className="xrk-timer-item xrk-f xrk-ac xrk-jc" data-testid={type} >
        {value}
      </View>
    )
  }

  return (
    <>
      <View className="xrk-timer-item xrk-f xrk-ac xrk-jc" data-testid={type} >
        {value}
      </View>
      <View className="xrk-timer-symbol">:</View>
    </>
  )
}
