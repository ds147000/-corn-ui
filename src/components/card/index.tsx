/*
 * @Description:
 * @Author: zhoulong.yang
 * @Date: 2021-06-07 12:03:34
 * @LastEditors: zhoulong.yang
 * @LastEditTime: 2021-06-07 15:43:55
 */
import React from 'react'
import { View, Text } from '@tarojs/components'

const Card: React.FC = () => {
  return (
    <View>
      <Text>卡片</Text>
    </View>
  )
}

export default Card
