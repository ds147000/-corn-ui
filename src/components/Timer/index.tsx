/* eslint-disable no-magic-numbers */
import React, { useCallback, useEffect, useMemo, useState } from 'react'
import { View } from '@tarojs/components'
import type { ViewProps } from '@tarojs/components/types/View'
import ClassNames  from 'classnames'
import { DateToTimestamp, fixNumber } from '../../utils'
import { SecondSchedulerDestotry, MinuteSchedulerApp, SecondSchedulerApp } from './SecondScheduler'
import { DefualtRenderItem } from './render'

const DAY_TIMESTAMP = 86400000
const MIN_TIMESTAMP = 60000
const HOUS_TIMESTAMP = 3600000

type TimeValue = {
  d: string
  h: string
  m: string
  s: string
}

export type TimeRenderItem = (timeType: 'day' | 'hous' | 'min' | 'sec', value: string, last?: boolean) => JSX.Element

export interface TimerProps extends ViewProps {
  /** 倒计时开始时间，可以是毫秒时间戳、日期字符串、日期对象 */
  startTime: number | string | Date
  /** 倒计时结束时间，可以是毫秒时间戳、日期字符串、日期对象 */
  endTime: number | string | Date
  /** 是否使用填充风格 */
  fill?: boolean
  /** 当剩余时间发现改变 */
  onChange?(TimeLeft: number): void
  /** 自定义返回时间项的渲染函数 */
  renderItem?: TimeRenderItem
}


const Timer: React.FC<TimerProps> = ({ startTime, endTime, onChange, renderItem, fill, className }) => {
  const [ timeValue, setTimeValue ] = useState<TimeValue>({ d: '', h: '', m: '', s: '' })

  const _class = useMemo(() => ClassNames(
    'xrk-timer xrk-if xrk-ac xrk-jc',
    fill && 'xrk-timer-fill',
    className
  ), [ fill, className ])

  const getTimeLeft = useCallback((): number => {
    if (endTime === startTime)
      return 0

    const start = DateToTimestamp(startTime)
    const end = DateToTimestamp(endTime)

    return end - start
  }, [ endTime, startTime ])

  useEffect(() => {
    let _time = getTimeLeft()

    if (_time <= 0) {
      setTimeValue({ s: '0', m: '0', h: '0', d: '' })
      return
    }

    let minDestory: SecondSchedulerDestotry, secDestory : SecondSchedulerDestotry

    // 销毁函数
    function runMinDestroyed(): void {
      if (minDestory) minDestory()
    }
    function runSecDestroyed(): void {
      if (secDestory) secDestory()
    }
    function destory(): void {
      runSecDestroyed()
      runMinDestroyed()
    }

    // 计算时间
    function runTime(interval: number): void {
      _time -= interval
      if (interval) onChange?.(_time)

      const s = fixNumber((_time % MIN_TIMESTAMP / 1000) | 0, 2)
      const m = fixNumber((_time % HOUS_TIMESTAMP / MIN_TIMESTAMP) | 0, 2)
      const h = fixNumber((_time % DAY_TIMESTAMP / HOUS_TIMESTAMP) | 0, 2)
      const d = (_time / DAY_TIMESTAMP) | 0
      const data = { s, m, h, d: d ? d + '' : '' }
      setTimeValue(data)

      // 倒计时结束销毁
      if (_time <= 0) {
        destory()
        return
      }


      // 大于一天半个小时，分钟调度
      if (_time > DAY_TIMESTAMP + (HOUS_TIMESTAMP / 2)) {
        runSecDestroyed() // 取消调度
        minDestory = MinuteSchedulerApp.add(runTime)
        return
      }

      runMinDestroyed() // 取消分钟调度
      secDestory = SecondSchedulerApp.add(runTime)
    }

    runTime(0)

    // eslint-disable-next-line consistent-return
    return destory
  }, [ getTimeLeft, onChange ])

  const _renderItem = renderItem || DefualtRenderItem
  return (
    <View className={_class}>
      {Boolean(timeValue.d) && (_renderItem('day', timeValue.d))}
      {_renderItem('hous', timeValue.h)}
      {_renderItem('min', timeValue.m, timeValue.d !== '')}
      {!timeValue.d && (_renderItem('sec', timeValue.s, timeValue.d === ''))}
    </View>
  )
}


export default Timer
