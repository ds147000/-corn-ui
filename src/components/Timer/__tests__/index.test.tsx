/* eslint-disable no-magic-numbers */
import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { View } from '@tarojs/components'
import { act } from 'react-dom/test-utils'
import Timer, { TimeRenderItem } from '../index'

beforeEach(() => {
  jest.useFakeTimers()
})

afterEach(() => {
  jest.useFakeTimers()
})

describe('comps/Timer', () => {

  test('基本使用', async () => {
    const startTime = 1500000000
    const endTime = startTime + 60000

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('01'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('基本使用 => fill', async () => {
    const startTime = 1500000000
    const endTime = startTime + 60000

    const screen = render(<Timer startTime={startTime} endTime={endTime} fill />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('01'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('基本使用 小时', async () => {
    const startTime = 1500000000
    const endTime = startTime + 3600000

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('01'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('基本使用 天', async () => {
    const startTime = 1500000000
    const endTime = startTime + (3600000 * 24)

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')?.innerHTML).toBe('1天'))
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('sec')).toBeNull())
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('基本使用 秒', async () => {
    const startTime = 1500000000
    const endTime = startTime + 3000

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('03'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('日期字符串', async () => {
    const startTime = '2021-10-01 12:00:00'
    const endTime = '2021-10-01 12:01:00'

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('01'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('Date对象', async () => {
    const startTime = new Date('2021-10-01 12:00:00')
    const endTime = new Date('2021-10-01 12:01:00')

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('01'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('分', async () => {
    const startTime = 1500000000
    const endTime = startTime + 120000

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('02'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
  })

  test('时', async () => {
    const startTime = 1500000000
    const endTime = startTime + (1000 * 60 * 60 * 23)

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')).toBeNull())
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('23'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('sec')?.innerHTML).toBe('00'))
  })

  test('天', async () => {
    const startTime = 1500000000
    const endTime = startTime + (1000 * 60 * 60 * 47)

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.queryByTestId('day')?.innerHTML).toBe('1天'))
    await waitFor(() => expect(screen.queryByTestId('hous')?.innerHTML).toBe('23'))
    await waitFor(() => expect(screen.queryByTestId('min')?.innerHTML).toBe('00'))
    await waitFor(() => expect(screen.queryByTestId('sec')).toBeNull())
  })

  test('回调onChange', async () => {
    const startTime = 1500000000
    const endTime = startTime + 60000
    const onChange = jest.fn()

    render(<Timer startTime={startTime} endTime={endTime} onChange={onChange} />)

    act(() => {
      jest.advanceTimersByTime(1000)
    })
    act(() => {
      jest.advanceTimersByTime(1000)
    })

    await waitFor(() => expect(onChange).toHaveBeenCalledTimes(2))
    await waitFor(() => expect(onChange.mock.calls[0][0]).toBe(59000))
    await waitFor(() => expect(onChange.mock.calls[1][0]).toBe(58000))
  })

  test('过期', async () => {
    const startTime = 1500000000
    const endTime = startTime

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('1秒后过期', async () => {
    const startTime = 1500000000
    const endTime = startTime + 1000

    const screen = render(<Timer startTime={startTime} endTime={endTime} />)

    act(() => {
      jest.advanceTimersByTime(1000)
      jest.advanceTimersByTime(1000)
      jest.advanceTimersByTime(1000)
    })

    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('自定义渲染像', async () => {
    const startTime = 1500000000
    const endTime = startTime + 1000

    const renderItem: TimeRenderItem = (type, value) => {
      return (
        <View>
          {type}={value}
        </View>
      )
    }
    const screen = render(<Timer startTime={startTime} endTime={endTime} renderItem={renderItem} />)

    await waitFor(() => expect(screen.getByText('hous=00')))
    await waitFor(() => expect(screen.getByText('min=00')))
    await waitFor(() => expect(screen.getByText('sec=01')))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })
})
