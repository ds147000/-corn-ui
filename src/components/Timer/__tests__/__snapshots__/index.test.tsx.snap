// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comps/Timer 1秒后过期 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer Date对象 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      01
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer 基本使用 => fill 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc xrk-timer-fill"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      01
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer 基本使用 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      01
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer 基本使用 天 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-f xrk-ac xrk-jc"
      data-testid="day"
    >
      1
      天
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer 基本使用 小时 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      01
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer 基本使用 秒 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      03
    </div>
  </div>
</div>
`;

exports[`comps/Timer 日期字符串 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      00
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      01
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      00
    </div>
  </div>
</div>
`;

exports[`comps/Timer 自定义渲染像 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class=""
    >
      hous
      =
      00
    </div>
    <div
      class=""
    >
      min
      =
      00
    </div>
    <div
      class=""
    >
      sec
      =
      01
    </div>
  </div>
</div>
`;

exports[`comps/Timer 过期 1`] = `
<div>
  <div
    class="xrk-timer xrk-if xrk-ac xrk-jc"
  >
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="hous"
    >
      0
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="min"
    >
      0
    </div>
    <div
      class="xrk-timer-symbol"
    >
      :
    </div>
    <div
      class="xrk-timer-item xrk-f xrk-ac xrk-jc"
      data-testid="sec"
    >
      0
    </div>
  </div>
</div>
`;
